src/main/scala/mesosphere/marathon/core/flow/FlowModule.scala

```diff
@@ -24,12 +27,23 @@ class FlowModule(leadershipModule: LeadershipModule) {
     */
   def reviveOffersWhenOfferMatcherManagerSignalsInterest(
     clock: Clock, conf: ReviveOffersConfig,
-    offersWanted: Observable[Boolean], driverHolder: MarathonSchedulerDriverHolder): Unit = {
-    lazy val reviveOffersActor = ReviveOffersActor.props(
-      clock, conf,
-      offersWanted, driverHolder
-    )
-    leadershipModule.startWhenLeader(reviveOffersActor, "reviveOffersWhenWanted")
+    marathonEventStream: EventStream,
+    offersWanted: Observable[Boolean],
+    driverHolder: MarathonSchedulerDriverHolder): Option[OfferReviver] = {
```

Comment:
shouldn't we change the method name to reflect that this now returns an Option[OfferReviver]?